lib/luminork-server/src/search/query.rs

```diff
@@ -68,34 +68,18 @@ impl SearchTerm {
         // TODO handle international case comparison as well. Library? icu? case_sensitive_string?
         match self {
             SearchTerm::Match(term) => {
-                // Rust doesn't have a `contains_ignore_ascii_case()` function, so we do it
-                // ourselves by checking if the any subset of bytes in the value *equal* the
-                // query string (ignoring case).
-                let term = term.as_bytes();
-                let value = value.as_bytes();
-                value
-                    .windows(term.len())
-                    .any(|window| window.eq_ignore_ascii_case(term))
-            }
-            SearchTerm::Exact(term) => term.eq_ignore_ascii_case(value),
-            SearchTerm::StartsWith(term) => {
-                // Rust doesn't have a `starts_with_ignore_ascii_case()` function, so we do it
-                // ourselves by checking if the first N bytes of the value *equal* the query string
-                // (ignoring case).
-                //
-                // We use as_bytes() here because s could potentially have multi-byte UTF-8
-                // characters. If you ask whether term="€" (3 bytes) starts with value="a" (1 byte),
-                // term[..value.len()] would panic because it would try to slice in the middle of
-                // the multi-byte character. Using as_bytes() avoids that, but still correctly
-                // checks equality, because it's a byte-by-byte comparison.
-                let term = term.as_bytes();
-                let value = value.as_bytes();
-                if value.len() >= term.len() {
-                    term.eq_ignore_ascii_case(&value[..term.len()])
-                } else {
-                    false
+                // If we have a*b*c, find a, then b, then c ...
+                let mut value = value;
+                for part in term.split('*') {
```

Comment:
To answer a question nobody asked: in Rust `split()` doesn't allocate, it's an iterator that scans through the string.